### ISSUE TITLE
Error on is_sorted & is_sorted_until

### ISSUE THREAD

**Issue by samchon** [samchon/tstl#34]
### Summary
  - **TSTL Version**: 2.1.2
  - **Expected behavior**: `is_sorted` & `is_sorted_until` requires only comparator like `less`
  - **Actual behavior**: They require not only comparator but also predicator (`equal_to`)

### Code occuring the bug
Functions `is_sorted` and `is_sorted_until` returns invalid value when target elements are not atomic values but custom objects that have not defined the predicator (`Object.equals`) function.

```typescript
import std = require("tstl");

function comparator(x: std.IPointer<number>, y: std.IPointer<number>): boolean
{
    return x.value < y.value;
}
function main(): void
{
    let pointers: std.Vector<std.IPointer<number>> = new std.Vector();
    for (let i: number = 0; i < 100; ++i)
        pointers.push_back({ value: std.randint(0, 10) });

    std.sort(pointers.begin(), pointers.end(), comparator);
    console.log( std.is_sorted(pointers.begin(), pointers.end(), comparator) );
        // MUST BE true
        // BUT PRINTS false
}
main();
```

**Comment by samchon** [comment]
https://github.com/samchon/tstl/blob/0aadfd3351b88fe13da4eedf8ce8bec54177d53b/src/algorithm/sorting.ts#L227

Functions `is_sorted` and `is_sorted_until` are using `equal_to` function. However, those functions do not provide the matched predicator function. The `equal_to` must be replaced to `!comp(x, y) && !comp(y, x)`.

**Comment by samchon** [comment]
Fixed on v2.1.3 patch
